components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3609,23 +3538,13 @@ where
             && self.fsm.skip_split_count < self.region_split_skip_max_count()
         {
             self.fsm.skip_split_count += 1;
-            return false;
+            self.register_split_region_check_tick();
+            return;
         }
         self.fsm.skip_split_count = 0;
 
-        let task =
-            SplitCheckTask::split_check(self.fsm.peer.region().clone(), true, CheckPolicy::Scan);
-        if let Err(e) = self.ctx.split_check_scheduler.schedule(task) {
-            error!(
-                "failed to schedule split check";
-                "region_id" => self.fsm.region_id(),
-                "peer_id" => self.fsm.peer_id(),
-                "err" => %e,
-            );
-        }
-        self.fsm.peer.size_diff_hint = 0;
-        self.fsm.peer.compaction_declined_bytes = 0;
-        true
+        self.fsm.peer.schedule_check_split(self.ctx);
+        self.register_split_region_check_tick();
```

Comment:
I suggest moving `if self.fsm.peer.has_calculated_region_size...` after the `is_leader` check and after that call `self.register_split_region_check_tick();`.
The condition `if !self.ctx.cfg.hibernate_regions {..}` can be removed.

Reply:
why `if !self.ctx.cfg.hibernate_regions {..}` can be removed

Reply:
Since the behavior can be the same whatever `hibernate_regions` is true or not. 
In the original code, the tick is registered each time even when this peer is a follower/learner which is very wasteful.